src/pybind/mgr/cephadm/serve.py

```diff
@@ -891,16 +891,58 @@ def remove_conflicting_daemons_for_service(self, spec: ServiceSpec) -> Tuple[boo
                     break
         return (r, hosts_altered)
 
-    def deploy_and_remove_daemons_by_service(self, spec: ServiceSpec) -> bool:
+    def handle_slot_names_and_rank_map_for_service(self, spec: ServiceSpec) -> List[DaemonPlacement]:
         service_type = spec.service_type
         service_name = spec.service_name()
-        svc = service_registry.get_service(service_type)
-        daemons = self.mgr.cache.get_daemons_by_service(service_name)
         slots_to_add = self.mgr.daemon_deploy_queue.get_queued_daemon_placements_by_service(spec.service_name())
         daemons_to_remove = self.mgr.daemon_removal_queue.get_queued_daemon_descriptions_by_service(spec.service_name())
+        svc = service_registry.get_service(service_type)
+        daemons = self.mgr.cache.get_daemons_by_service(service_name)
         rank_map = None
         if svc.ranked(spec):
             rank_map = self.mgr.spec_store[spec.service_name()].rank_map or {}
+        # assign names
+        for i in range(len(slots_to_add)):
```

Comment:
```
for idx, slot_to_add enumerate(slots_to_add):
```
is more pythonic. I do see below that the list is being mutated but I would suggest building a new list while iteratring over the old is also more pythonic.
This code feels a bit C-ish to me.